python/src/lib/workbook_validator.py

```diff
@@ -8,9 +8,28 @@
     CoverSheetRow,
     LogicSheetVersion,
     SubrecipientRow,
+    BaseProjectRow,
+    BaseModel
 )
 
-type Errors = List[str]
+type Errors = List[WorkbookError]
+
+LOGIC_SHEET = "Logic"
+COVER_SHEET = "Cover"
+PROJECT_SHEET = "Project"
+SUBRECIPIENTS_SHEET = "Subrecipients"
+
+class WorkbookError:
+    message: str
+    row: str
+    col: str
+    tab: str
```

Comment:
would recommend adding another field here to capture the `field_name`. We aren't using this yet in the front-end but I think it may be useful in the future to display it.